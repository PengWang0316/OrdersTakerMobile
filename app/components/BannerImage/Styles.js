import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  rootView: {
    width: '100%',
    height: '100%',
  },
});
