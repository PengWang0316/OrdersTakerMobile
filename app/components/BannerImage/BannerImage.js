import React, { Component } from 'react';
import { View, Image, Dimensions } from 'react-native';
import { connect } from 'react-redux';

import Styles from './Styles';
import { fetchBasicInformation } from '../../actions/BasicInformationActions';

type Props = {
  basicInformation: Object,
  fetchBasicInformation: Function,
};

/**
 * The component to show the banner image.
 */
export class BannerImage extends Component<Props> {
  /**
   * Calling the fetchBasicInformation action if the Redux state has not been filled yet.
   * @param {object} props contains prop values.
   */
  constructor(props) {
    super(props);
    const dimensions = Dimensions.get('window');
    this.imageWidth = dimensions.width;
    this.imageHeight = Math.floor(dimensions.height * 0.25);
    if (!props.basicInformation) props.fetchBasicInformation();
  }

  /**
   * The render method
   * @return {jsx} Return jsx.
   */
  render() {
    const { basicInformation } = this.props;
    return (
      <View style={Styles.rootView}>
        {basicInformation && (
          <Image
            source={{ uri: basicInformation.banners[0].url }}
            style={{ width: this.imageWidth, height: this.imageHeight }}
          />
        )}
      </View>
    );
  }
}
/* istanbul ignore next */
const mapStateToProps = ({ basicInformation }) => ({
  basicInformation,
});
/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  fetchBasicInformation: () => dispatch(fetchBasicInformation()),
});
export default connect(mapStateToProps, mapDispatchToProps)(BannerImage);
