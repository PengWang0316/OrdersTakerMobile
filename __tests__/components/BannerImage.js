import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { BannerImage } from '../../app/components/BannerImage/BannerImage';

jest.mock('react-native', () => ({
  View: () => 'View',
  Image: () => 'Image',
  Dimensions: ({
    get: jest.fn().mockReturnValue({ width: 100, height: 200 }),
  }),
  StyleSheet: { create: jest.fn().mockReturnValue({ rootView: {} }) },
}));

describe('BannerImage', () => {
  const defaultProps = {
    basicInformation: { banners: [{ url: 'url' }] },
    fetchBasicInformation: jest.fn(),
  };
  const getShallowComponent = (props = defaultProps) => shallow(<BannerImage {...props} />);

  test('Constructor bisicInformation does not equal null', () => {
    const { Dimensions } = require('react-native');
    const component = getShallowComponent();
    expect(defaultProps.fetchBasicInformation).not.toHaveBeenCalled();
    expect(Dimensions.get).toHaveBeenCalledTimes(1);
    expect(Dimensions.get).toHaveBeenLastCalledWith('window');
    expect(component.instance().imageWidth).toBe(100);
    expect(component.instance().imageHeight).toBe(50);
  });

  test('Constructor bisicInformation equals null', () => {
    getShallowComponent({ ...defaultProps, basicInformation: null });
    expect(defaultProps.fetchBasicInformation).toHaveBeenCalledTimes(1);
  });

  test('Snapshot has basicInformation', () => expect(renderer.create(<BannerImage {...defaultProps} />).toJSON()).toMatchSnapshot());
  test('Snapshot does not have basicInformation', () => expect(renderer.create(<BannerImage {...{ ...defaultProps, basicInformation: null }} />).toJSON()).toMatchSnapshot());
});
