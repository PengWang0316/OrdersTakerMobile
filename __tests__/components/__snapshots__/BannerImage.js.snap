// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerImage Snapshot does not have basicInformation 1`] = `"View"`;

exports[`BannerImage Snapshot has basicInformation 1`] = `"View"`;
